pandas/core/groupby/groupby.py

```diff
@@ -2326,8 +2326,9 @@ def head(self, n=5):
         """
         Return first n rows of each group.
 
-        Essentially equivalent to ``.apply(lambda x: x.head(n))``,
-        except ignores as_index flag.
+        Similar to ``.apply(lambda x: x.head(n))``, but it returns a subset of rows
+        from the original DataFrame with original index preserved (``as_index`` flag
```

Comment:
I would also explicitly mention that the original *order* is preserved

Reply:
fixed in latest commit